### PR TITLE
Migrate this.ref.dotMenu to functional

### PR DESCRIPTION
**Pull request by egrinberg**
#### Summary

Migrate this.ref.dotMenu to React.createRef

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/15845

JIRA ticket: https://mattermost.atlassian.net/browse/MM-29406

### CODE REVIEW

**Review by nickmisasi — COMMENTED** [review]
Change looks good, thanks @egrinberg!

Will need to fix the linter issues - it prefers `{foo}` over `{ foo }`

**Comment by egrinberg** [comment]
@nickmisasi Thank you for the feedback. The formatting is fixed

File: components/rhs_root_post/rhs_root_post.jsx [hunk 1]
```diff
@@ -342,7 +343,7 @@ class RhsRootPost extends React.PureComponent {
         if (!isPostDeleted && this.props.post.type !== Constants.PostTypes.FAKE_PARENT_DELETED) {
             dotMenuContainer = (
                 <div
-                    ref='dotMenu'
+                    ref={this.dotMenuRef}
```
Comment (nickmisasi) [inline components/rhs_root_post/rhs_root_post.jsx:346]: I don't think this is defined? It think you need either `this.getDotMenuRef` or `this.dotMenu` here.

**Review by nickmisasi — CHANGES_REQUESTED** [review]

File: components/rhs_root_post/rhs_root_post.jsx (hunk 1 above)
Comment (egrinberg) [reply components/rhs_root_post/rhs_root_post.jsx:346]: Yes, I think you right. I will fix it.

File: components/rhs_root_post/rhs_root_post.jsx [hunk 2]
```diff
@@ -86,6 +86,7 @@ class RhsRootPost extends React.PureComponent {
         };
 
         this.postHeaderRef = React.createRef();
+        this.dotMenu = React.createRef();
```
Comment (Willyfrog) [inline components/rhs_root_post/rhs_root_post.jsx:89]: minor change, can we rename to dotMenuRef for consistency with other PRs
```suggestion
        this.dotMenuRef = React.createRef();
```

File: components/rhs_root_post/rhs_root_post.jsx [hunk 3]
```diff
@@ -342,7 +343,7 @@ class RhsRootPost extends React.PureComponent {
         if (!isPostDeleted && this.props.post.type !== Constants.PostTypes.FAKE_PARENT_DELETED) {
             dotMenuContainer = (
                 <div
-                    ref='dotMenu'
+                    ref={this.getDotMenuRef}
```
Comment (Willyfrog) [inline components/rhs_root_post/rhs_root_post.jsx:346]: not really a problem, but you can link the ref directly. the getter is mostly for anything outside the component
```suggestion
                    ref={this.dotMenuRef}
```

**Review by Willyfrog — CHANGES_REQUESTED** [review]
Thanks for working on this PR! I added a couple of minor changes, can you also remove the comments (on top and bottom of the file) that removes the eslint rule for string-refs?

File: components/rhs_root_post/rhs_root_post.jsx (hunk 2 above)
Comment (egrinberg) [reply components/rhs_root_post/rhs_root_post.jsx]: the change is done

File: components/rhs_root_post/rhs_root_post.jsx (hunk 3 above)
Comment (egrinberg) [reply components/rhs_root_post/rhs_root_post.jsx]: This is also changed.

**Review by Willyfrog — CHANGES_REQUESTED** [review]
looks good, once you fix the linting errors I think it is good to go :D
thanks!

**Review by Willyfrog — APPROVED** [review]
LGTM! thanks for fixing all those 👍

**Review by nickmisasi — APPROVED** [review]
Everything looks good to me! Thanks @egrinberg!

**Review by mkraft — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thank you @egrinberg
Tested, looks good to merge.
- Verified 3-dot menu on the rhs root post - no change in functionality.
@Willyfrog Can you please help merge? Thanks 🙂

**Pull request merged**